numpy/linalg/linalg.py

```diff
@@ -2112,6 +2112,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
     """
     x = asarray(x)
 
+    if not issubclass(x.dtype.type, floating):
+        x = x.astype(float)
```

Comment:
`np.float` is just an alias to `float` (i.e. `np.float is float` will be `True`). In fact, I *believe* those are planned to be removed, but I can't find the issue right now.